templates/components/form.jsx

```diff
@@ -0,0 +1,98 @@
+var React = require('react/addons');
+var ValidationMixin = require('react-validation-mixin');
+var Joi = require('joi');
+
+var Signup = React.createClass({
+  propTypes: {
+    fields: React.PropTypes.array.isRequired,
+    btnTxt: React.PropTypes.string.isRequired
+  },
+  statics: {
+      'phoneLabels': {
+        'username': 'phone-label-username',
+        'password': 'phone-label-password',
+        'email': 'phone-label-email'
+      }
+  },
+  mixins: [ValidationMixin, React.addons.LinkedStateMixin],
+  validatorTypes:  {
+    firstName: Joi.string().required().label('First Name'),
+    lastName: Joi.string().allow(null).label('Last Name'),
+    email: Joi.string().email().label('Email Address'),
+    username:  Joi.string().alphanum().min(3).max(30).required().label('Username'),
+    password: Joi.string().regex(/[a-zA-Z0-9]{3,30}/).label('Password'),
+    verifyPassword: Joi.any().valid(Joi.ref('password')).required().label('Password Confirmation')
+  },
+  getInitialState: function() {
+    return {
+      firstName: null,
+      lastName: null,
+      email: null,
+      username: null,
+      password: null,
+      verifyPassword: null,
+      feedback: null
+    };
+  },
+  render: function() {
+     var that = this;
+     var fields = Object.keys(this.props.fields).map(function(key,i) {
```

Comment:
we can pull the function we pass into `map` as a component function instead (e.g. `mapfunction`) and then call it as `...map(this.mapfunction)`, and then because React does `this` rebinding, we don't need the self=this variable. It can stay "this" inside the mapping function